mailbox/store/src/test/java/org/apache/james/mailbox/store/extractor/JsoupTextExtractorTest.java

```diff
@@ -81,19 +81,18 @@ void extractContentShouldReturnEmptyWhenNullContentType() throws Exception {
             .isEqualTo(ParsedContent.empty());
     }
 
-    @Disabled("JAMES-3044 java.io.IOException: Input is binary and unsupported")
     @Test
-    void extractContentShouldNotThrowWhenContainingNullCharacters() {
+    void extractContentShouldNotThrowWhenContainingNullCharacters() throws Exception {
         InputStream inputStream = textContentWithManyNullCharacters();
 
-        assertThatCode(() -> textExtractor.extractContent(inputStream, TEXT_HTML_CONTENT_TYPE))
-            .doesNotThrowAnyException();
+        assertThat(textExtractor.extractContent(inputStream, TEXT_HTML_CONTENT_TYPE)
+                .getTextualContent())
+            .hasValueSatisfying(text -> assertThat(text)
+                .contains(HTML_TEXT_CONTENT));
     }
 
     private InputStream textContentWithManyNullCharacters() {
-        String htmlTextContent = "HTML pages can include a lot of null '\0' character. But still expecting the content can be parsed." +
-            "Jsoup 1.21.1 thinks a file containing more than 10 null characters can be a binary file";
-        byte[] htmlBytesContent = htmlTextContent.getBytes(StandardCharsets.UTF_8);
+        byte[] htmlBytesContent = HTML_TEXT_CONTENT.getBytes(StandardCharsets.UTF_8);
         byte[] nullCharacters = {'\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0', '\0'};
```

Comment:
you can simply add `"\0\0\0\0\0\0\0\0\0\0"` at this end of the string instead of this unneeded byte array concatenation.

Reply:
:trollface: that's the trick man. Okay, I need an InputStream to pass to Jsoup, to have the InputStream, I need bytes from the string (`myString.getBytes("UTF-8")`) => unlucky, this UTF-8 encoding had ignored my '\0' characters, maybe it doesn't match with the UTF-8 encoding rules. 

The '\0' should be in a separated bytes without any String encoding/decoding. Otherwise, my '\0' will disappear.

Reply:
it seems wrong, I tested and it works fine for me

Reply:
this commit https://github.com/rouazana/james-project/commit/0d224ad465962bfdd3965f64b40c66d46a975f91 shows a red test

I would guess that you did not refresh maven in your IDE, making it use the old library, no?

Reply:
OH sorry. Before come up with this thing, I had download the HTML file reported in the Jsoup issue. And save it as a text file. Every time the test loading this file, I haven't encountered any issue of Jsoup. While using Jsoup parsing the associated URL could breaks. So, I guess UTF-8 encoding did something underneath the tree.

Thank you!